### ISSUE TITLE
handlebars to react converter

### ISSUE THREAD

**Issue by faceyspacey** [timbrandin/blaze-react#42]
https://github.com/stevenvachon/handlebars-react

that may help you. 

here's another, but for more general templating:

http://wix.github.io/react-templates/

**Comment by timbrandin** [comment]
Nice! Thx

**Comment by timbrandin** [comment]
Though it uses the handlebars-html-parser, which gives a lot of overhead.

**Comment by timbrandin** [comment]
@faceyspacey an update after some hacking this weekend.

I stumbled upon the new feature Proxy in ES6, have you tried it?
It basically gives us the possibility to do `{context.var.param}` without a helper, the reason I did one to start with is that react requires all variables used to be predefined, which isn't so helpful in a template actually, it should fail gracefully! Though Proxy is not yet ready using babel, it needs to be added by the browsers, but anyhow it inspired me in a new lookup function that is instead recursive and works really well. I'm about to add it to the BlazeReact class, but I can't wait for the Proxy method as it is such a nicer method to do this. I'll share it when I've committed it!

Regarding the template parsing, I've realized we need to use XRegExp to be able to match recursive patterns and nested template logic.

One mayor issue with react though is that we have to sprinkle in `<span>` wherever one is missing an element wrapping a block helper such as #each, #if and #unless :/ I hoped to find a solution to overcome this, but haven't been able to.
